File: backend/src/main/java/com/rgt/restaurant/model/Menu.java
package com.rgt.restaurant.model;

import java.math.BigDecimal;
import java.util.List;
import java.util.Map;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Menu {
    private String id;
    private String name;
    private String description;
    private BigDecimal price;
    private Category category;
    private String imageUrl;
    private boolean isBestSeller;
    private boolean isAvailable;
    private int preparationTime;
    private int spicyLevel;
    private List<String> allergens;
    private Map<String, Integer> nutritionalInfo;

    public Menu(
        String id,
        String name,
        String description,
        double price,
        Category category,
        String imageUrl,
        boolean isBestSeller,
        boolean isAvailable,
        int preparationTime,
        int spicyLevel,
        List<String> allergens,
        Map<String, Integer> nutritionalInfo
    ) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = BigDecimal.valueOf(price);
        this.category = category;
        this.imageUrl = imageUrl;
        this.isBestSeller = isBestSeller;
        this.isAvailable = isAvailable;
        this.preparationTime = preparationTime;
        this.spicyLevel = spicyLevel;
        this.allergens = allergens;
        this.nutritionalInfo = nutritionalInfo;
    }
} 

File: customer/src/types/Menu.ts
export interface Category {
  id: string;
  name: string;
  description?: string;
}

export interface Menu {
  id: string;
  name: string;
  description: string;
  price: number;
  category: {
    id: string;
    name: string;
    description: string;
  };
  imageUrl: string;
  preparationTime: number;
  spicyLevel: number;
  allergens: string[];
  nutritionalInfo: {
    calories: number;
    protein: number;
    carbs: number;
    fats: number;
  };
  bestSeller: boolean;
  available: boolean;
}

export interface MenuRequest {
  name: string;
  description: string;
  price: number;
  imageUrl?: string;
  categoryId: string;
  preparationTime?: number;
  spicyLevel?: number;
  allergens?: string[];
} 